PowerFGT/Public/cmdb/webfilter/urlfilter.ps1

```diff
@@ -0,0 +1,441 @@
+# Copyright 2022, Cédric Moreau <moreaucedric0 dot gmail dot com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+function Add-FGTWebfilterUrlfilter {
+
+    <#
+        .SYNOPSIS
+        Add a FortiGate URL Filter
+
+        .DESCRIPTION
+        Add a FortiGate URL Filter
+
+        .EXAMPLE
+        Add-FGTWebfilterUrlfilter -name myURL1 -id 1 -name MyURL -comment "Added by PowerFGT"
+
+        Add URL Filter object named MyURL with comment
+
+        .EXAMPLE
+        Add-FGTWebfilterUrlfilter -name myURL1 -id 1 -name MyURL -comment "Added by PowerFGT" -url_id 1 -url_type simple -url powerfgt.com -action allow -status enable
+
+        Add URL Filter object named MyURL with an url (URL : powerfgt.com, type : simple, action : allow, status : enable)
+
+        .EXAMPLE
+        Add-FGTWebfilterUrlfilter -name myURL1 -id 1 -name MyURL -comment "Added by PowerFGT" -url_id 1 -url_type wildcard -url *powerfgt.com -action block -status enable
+
+        Add URL Filter object named MyURL with an url (URL : *powerfgt.com, type : wildcard, action : block, status : enable)
+
+    #>
+
+    Param(
+        [Parameter (Mandatory = $false)]
+        [string]$id,
+        [Parameter (Mandatory = $true)]
+        [string]$name,
+        [Parameter (Mandatory = $false)]
+        [string]$url_id,
+        [Parameter (Mandatory = $false)]
+        [string]$url_type,
+        [Parameter (Mandatory = $false)]
+        [string]$url,
+        [Parameter (Mandatory = $false)]
+        [string]$action,
+        [Parameter (Mandatory = $false)]
+        [string]$status,
+        [Parameter (Mandatory = $false)]
+        [string]$exempt,
+        [Parameter (Mandatory = $false)]
+        [switch]$skip,
+        [Parameter(Mandatory = $false)]
+        [String[]]$vdom,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        $invokeParams = @{ }
+        if ( $PsBoundParameters.ContainsKey('skip') ) {
+            $invokeParams.add( 'skip', $skip )
+        }
+        if ( $PsBoundParameters.ContainsKey('vdom') ) {
+            $invokeParams.add( 'vdom', $vdom )
+        }
+
+        if ( Get-FGTWebfilterUrlfilter -connection $connection @invokeParams -name $name ) {
+            Throw "Already a URL profile object using the same name"
+        }
+
+        $uri = "api/v2/cmdb/webfilter/urlfilter"
+
+        $urlfilter = new-Object -TypeName PSObject
+
+        $urlfilter | add-member -name "name" -membertype NoteProperty -Value $name
+
+        $urlfilter | add-member -name "id" -membertype NoteProperty -Value $id
+
+        if ( $PsBoundParameters.ContainsKey('comment') ) {
+            $_urlfilter | add-member -name "comment" -membertype NoteProperty -Value $comment
+        }
+
+        $_entry = new-Object -TypeName PSObject
+
+        if ( $PsBoundParameters.ContainsKey('url_id') ) {
+            $_entry | add-member -name "id" -membertype NoteProperty -Value $url_id
```

Comment:
typo ? id => url_id ?

Reply:
id = ID of URL Filter profile
url_id = ID of URL inside that profile

So id =/= url_id